app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt

```diff
@@ -486,12 +486,17 @@ class RefreshController @Inject constructor(private val localSettings: LocalSett
     ): Set<Thread> {
 
         val threadsToUpsert = mutableSetOf<Thread>()
+        val existingMessages = folder.messages.associateByTo(mutableMapOf()) { it.uid }
 
         remoteMessages.forEach { remoteMessage ->
             scope.ensureActive()
 
-            val isMessageAlreadyAdded = folder.messages.firstOrNull { it == remoteMessage && !it.isOrphan() } != null
-            if (isMessageAlreadyAdded) {
+            // Get updated version of existing Message
+            val existingMessage = existingMessages[remoteMessage.uid]?.let {
+                if (it.isManaged()) it else MessageController.getMessage(it.uid, realm = this)
+            }
```

Comment:
Where do the messages that are unmanaged at this point come from again?

Reply:
From the `remoteMessage`.